### PR TITLE
Add timeline labels

### PR DESCRIPTION
**Pull request by rogaldh**
MEDIA-1351

### CODE REVIEW

**Comment by alexkonst** [comment]
Why Travis don't trigger build for branch?

**Comment by rogaldh** [comment]
@alexkonst, this would be in next PR. Build stage inside travis + `npm publish` looks great together.

File: package.json [hunk 1]
```diff
@@ -1,34 +1,36 @@
 {
-  "name": "react-hls",
-  "version": "2.1.1",
+  "name": "@netology-group/react-hls",
```
Comment (alexkonst) [inline package.json]: This repo is under `@foxford` org.

File: dist/Markers.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+Object.defineProperty(exports, "__esModule", {
+  value: true
+});
+exports.Markers = undefined;
+
+require('videojs-markers');
```
[the remark this answers is not in the document]
Comment (rogaldh) [reply dist/Markers.js]: That was left after rebase. All stuff with markers won't be there

[the remark this answers is not in the document]
Comment (alexkonst) [reply dist/Markers.js]: Currently we holding files from `dist` directory in repo. So you need to compile src (by running `npm run compile`) and commit files in `dist` directory.

File: package.json (hunk 1 above)
Comment (rogaldh) [reply package.json]: There is no @foxford for the moment. That's there for dev reasons mostly

File: package.json [hunk 3]
```diff
@@ -1,34 +1,36 @@
...
+  "version": "2.2.1",
```
[the remark this answers is not in the document]
Comment (rogaldh) [reply package.json]: Was incremented to 2.2.1 as there have been 2 commits.

File: src/control-bar/DisplayBarLabeled.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+  return {
+    timePosRatio,
+    offset: mouseTimeOffset,
+    timeWidth: labelWidthInTime,
+    posWidth: labelWidthInPos
+  }
+}
+
+const DisplayBarLabeled = function (props) {
```
[the remark this answers is not in the document]
Comment (rogaldh) [reply src/control-bar/DisplayBarLabeled.js]: typo :shrug:

File: src/control-bar/DisplayBarLabeled.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    mouseTime,
+    currentTime,
+    width = 4,
+    chapters = [],
+    classes = {}
+  } = props
+
+  if (!duration || !mouseTime.time) return null
+
+  const Labels = chapters.map((it) => {
```
[the remark this answers is not in the document]
Comment (rogaldh) [reply src/control-bar/DisplayBarLabeled.js]: For the moment `Labels` don't have sense outside `DisplayBarLabeled` and use some props also.

File: src/control-bar/DisplayBarLabeled.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    const {
+      timePosRatio, offset, posWidth: pxWidth, timeWidth
+    } = calcLabelDimensions(it.start, mouseTime, width)
+
+    const shiftLeft = it.start * 100 / duration
+    const isCursorAboveLabel = offset < timeWidth / 2
+    const isLabelPassed = it.start <= currentTime
+    const shouldColorLabers = timePosRatio / duration <= 1.5 * 1e-2
```
[the remark this answers is not in the document]
Comment (rogaldh) [reply src/control-bar/DisplayBarLabeled.js]: Say again

File: src/control-bar/DisplayBarLabeled.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    const styles = {
+      left: `${shiftLeft}%`,
+      position: 'absolute',
+      transform: 'translate(-50%)',
+      width: `${pxWidth}px`,
+      height: '.3em'
+    }
+
+    return <div
+      key={it.label}
```
[the remark this answers is not in the document]
Comment (rogaldh) [reply src/control-bar/DisplayBarLabeled.js]: Keys should be unique. Array indexes might be changed after sort.

File: src/control-bar/DisplayBarLabeled.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+      className={classNames('video-react-progress-labels', classes.item)}
+      style={!shouldColorLabers ? styles : {
+        ...styles,
+        backgroundColor: isCursorAboveLabel
+          ? '#06ff60' : (isLabelPassed ? '#ffec17' : '#ffec17')
```
[the remark this answers is not in the document]
Comment (rogaldh) [reply src/control-bar/DisplayBarLabeled.js]: That's for possible style tweak. Label displayed before current position lay on top of «played» progress bar.

File: dist/Markers.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rogaldh) [reply dist/Markers.js]: :\

File: dist/HLSSource.js [hunk 9]
```diff
@@ -24,6 +24,8 @@ function _possibleConstructorReturn(self, call) { if (!self) { throw new Referen
 
 function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }
 
+// eslint-disable-line import/no-webpack-loader-syntax
```
[the remark this answers is not in the document]
Comment (rogaldh) [reply dist/HLSSource.js]: :\

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (alexkonst) [reply package.json]: I only see 1 commit, that results in increasing minor version to 2.

[the remark this answers is not in the document]
Comment (rogaldh) [reply package.json]: It was rebased :\

File: dist/Markers.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rogaldh) [reply dist/Markers.js]: There is no point to store `/dist` since it will be available on NPM.

**Comment by rogaldh** [comment]
> So we use Travis for npm publish or for git workflow?? How i can be sure that tests passed on this branch?
https://github.com/foxford/react-hls/commit/c41997c406d6fb60793051d2d95de0b2434fdd76

File: package.json [hunk 10]
```diff
@@ -44,7 +46,6 @@
     "eslint-plugin-standard": "^3.0.1",
     "extract-text-webpack-plugin": "^2.1.0",
     "file-loader": "^0.11.1",
-    "react-dom": "^15.4.2",
```
Comment (alexkonst) [inline package.json]: This package used for rendering example.

Comment (rogaldh) [reply package.json]: Fixed

File: package.json [hunk 11]
```diff
@@ -44,8 +50,13 @@
...
+    "node-sass": "^4.9.0",
+    "rc-slider": "^7.0.8",
+    "react-dom": "^15.6.1",
     "react-hot-loader": "^1.3.1",
+    "redux": "^3.7.1",
```
Comment (alexkonst) [inline package.json]: This package not used.

File: src/Player.js [hunk 12]
```diff
@@ -84,7 +86,12 @@ class ReactPlayer extends Component {
         }
         <BigPlayButton position='center' />
         <ControlBar>
-          <VolumeMenuButton order={7} />
+          {hasChapters && <ProgressControlLabeled
```
Comment (alexkonst) [inline src/Player.js]: What if `hasChapters` equals to `false`? We can't see `ProgressControlLabeled` at all?

File: package.json (hunk 11 above)
Comment (rogaldh) [reply package.json]: Forgot about it. Start with `re` looks like smth for react 😄

File: src/Player.js (hunk 12 above)
Comment (rogaldh) [reply src/Player.js]: Original `ProgressControl` will be rendered.

**Pull request merged**